### PR TITLE
Fixes 14457 (VariantN assignment from subset VariantN)

### PR DESCRIPTION
**Pull request by jwhear**
https://issues.dlang.org/show_bug.cgi?id=14457

Currently VariantN allows assignment from its allowed types and from a an instance of the exact same type.  This expands that functionality to allow assignment from another VariantN that has a strict subset of allowed types.  Implements an opAssign overload that specializes on VariantN instantations that are not the current VariantN but whose AllowedTypes are a subset of the current variant's.

The new overload has an explanatory comment, but does not have a documentation comment as the original opAssign is not documented.

Adds an accompanying unittest.

Fix Issue 14457

### CODE REVIEW

**Comment by schuetzm** [comment]
Would it make sense to implement the same feature for construction?

File: std/variant.d [hunk 1]
```diff
@@ -660,6 +660,18 @@ public:
         return this;
     }
 
+    // Allow assignment from another variant which is a subset of this one
+    VariantN opAssign(T : VariantN!(Tsize, Types), size_t Tsize, Types...)(T rhs)
```
Comment (JakobOvrum) [inline std/variant.d]: nitpick; s/Tsize/tsize/

**Comment by JakobOvrum** [comment]
LGTM, but I agree on construction.

**Comment by jwhear** [comment]
@schuetzm Thanks, hadn't thought about the constructor.  Added.
@JakobOvrum Changed `Tsize` to `tsize`

**Comment by JakobOvrum** [comment]
Docs are a little thin, but LGTM. Will merge within a few days if there is no further activity.

**Comment by JakobOvrum** [comment]
Auto-merge toggled on

**Pull request merged**